src/pebble.py

```diff
@@ -62,6 +62,29 @@ def check_synapse_alive() -> ops.pebble.CheckDict:
     return check.to_dict()
 
 
+def validate_pebble_layer(
+    container: ops.model.Container, pebble_layer: ops.pebble.LayerDict
+) -> ops.pebble.LayerDict:
+    """Validate pebble layer.
+
+    For now, the only validation is remove LD_PRELOAD var from pebble layer
+    if libjemalloc not found.
+
+    Args:
+        container: Synapse container.
+        pebble_layer: Pebble layer.
+
+    Returns:
+        New pebble layer validated.
+    """
+    if not container.exists(synapse.LIBJEMALLOC_PATH):
+        logger.debug(
+            "%s not found, removing LD_PRELOAD environment variable", synapse.LIBJEMALLOC_PATH
+        )
```

Comment:
Do we really want this as a debug level message, rather than (for instance) warning? It seems like it's a pretty big red flag if this isn't installed since we're installing it in the rock, so we'd always expect it to be there?

Reply:
Thanks Niels, it's a good point. As discussed, this check was removed since is not done in any other part of the charm and we own the Rock.